File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem12_ContainerWithMostWater.java
package org.phoenix.leetcode.challenges;

public class Problem12_ContainerWithMostWater {
    public int maxArea(int[] height) {
        int end = height.length - 1;
        int start = 0;
        int maxArea = Integer.MIN_VALUE;
        while (start < end) {
            maxArea = Math.max(maxArea, Math.min(height[start], height[end]) * (end - start));

            if (height[start] < height[end]) {
                start++;
            } else {
                end--;
            }
        }
        return maxArea;
    }
}

File: JavaScript/leetcode75/src/main/Problem12_ContainerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let end = height.length - 1;
  let start = 0;
  let maxArea = Number.MIN_VALUE;
  while (start < end) {
    maxArea = Math.max(
      maxArea,
      Math.min(height[start], height[end]) * (end - start)
    );

    if (height[start] < height[end]) {
      start++;
    } else {
      end--;
    }
  }
  return maxArea;
};

module.exports = maxArea;
